### PR TITLE
[CWS] add an ebpfless CI job

### PR DESCRIPTION
**Pull request by safchain**
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/tests/module_tester.go [hunk 1]
```diff
@@ -966,24 +962,19 @@ func newTestModule(t testing.TB, macroDefs []*rules.MacroDefinition, ruleDefs []
...
 		} else {
-			wrapper, err := newDockerCmdWrapper(st.Root(), st.Root(), "ubuntu")
-			if err == nil {
-				cmdWrapper = newMultiCmdWrapper(wrapper, newStdCmdWrapper())
-			} else {
-				// docker not present run only on host
-				cmdWrapper = newStdCmdWrapper()
-			}
+			// docker not present run only on host
+			cmdWrapper = newStdCmdWrapper()
```
Comment (spikat) [inline pkg/security/tests/module_tester.go:973]: is this changes have to be removed/adpated? as they already were part of another PR (#22192)

File: pkg/security/tests/main_test.go [hunk 2]
```diff
@@ -11,14 +11,76 @@ package tests
...
+			}
+		}
+
+		if slices.Contains(exclude, t.Name()) {
+			t.Skip("test not available for ebpfless")
+		}
+
+		if !execTest {
+			t.Skip("test not available for ebpfless")
+		}
```
Comment (spikat) [inline pkg/security/tests/main_test.go:80]: nit: merge those ifs ? :)

File: pkg/security/tests/mkdir_test.go [hunk 3]
```diff
@@ -86,7 +88,7 @@ func TestMkdir(t *testing.T) {
 			return nil
 		}, func(event *model.Event, rule *rules.Rule) {
 			assertTriggeredRule(t, rule, "test_rule_mkdirat")
-			if !test.opts.staticOpts.enableEBPFLess {
+			if !ebpfLessEnabled {
 				assert.Equal(t, getInode(t, testatFile), event.Mkdir.File.Inode, "wrong inode")
```
Comment (spikat) [inline pkg/security/tests/mkdir_test.go:92]: could use the new helper here

File: pkg/security/tests/main_test.go [hunk 4]
```diff
@@ -11,14 +11,76 @@ package tests
...
+			"~TestProcess",
+			"~TestOpen",
+			"~TestUnlink",
+			"KillAction",
+			"TestRmdir",
+			"TestUnlink",
+			"TestRename",
+			"TestMkdir",
+			"TestUtimes",
+		}
```
Comment (YoannGh) [inline pkg/security/tests/main_test.go:51]: There's both `~TestUnlink` and `TestUnlink`. Should we remove `TestUnlink`  and only keep `~TestUnlink`?

File: pkg/security/tests/rmdir_test.go [hunk 5]
```diff
@@ -58,7 +60,7 @@ func TestRmdir(t *testing.T) {
 			return nil
 		}, func(event *model.Event, rule *rules.Rule) {
 			assert.Equal(t, "rmdir", event.GetType(), "wrong event type")
-			if !test.opts.staticOpts.enableEBPFLess {
+			if !ebpfLessEnabled {
 				assert.Equal(t, inode, event.Rmdir.File.Inode, "wrong inode")
```
Comment (spikat) [inline pkg/security/tests/rmdir_test.go:64]: here we could use the new inode assert helper

File: pkg/security/tests/rmdir_test.go [hunk 6]
```diff
@@ -90,7 +92,7 @@ func TestRmdir(t *testing.T) {
...
```
Comment (spikat) [inline pkg/security/tests/rmdir_test.go:96]: here we could use the new inode assert helper

File: pkg/security/tests/rmdir_test.go [hunk 7]
```diff
@@ -103,9 +105,8 @@ func TestRmdir(t *testing.T) {
 	})
 
 	t.Run("unlinkat-io_uring", func(t *testing.T) {
-		if test.opts.staticOpts.enableEBPFLess {
-			t.Skip("io_uring not supported")
-		}
+		SkipIfNotAvailable(t)
```
Comment (spikat) [inline pkg/security/tests/rmdir_test.go:108]: This test is not listed on the exclude list

File: pkg/security/tests/unlink_test.go [hunk 8]
```diff
@@ -53,7 +55,7 @@ func TestUnlink(t *testing.T) {
 			return nil
 		}, func(event *model.Event, rule *rules.Rule) {
 			assert.Equal(t, "unlink", event.GetType(), "wrong event type")
-			if !test.opts.staticOpts.enableEBPFLess {
+			if !ebpfLessEnabled {
 				assert.Equal(t, inode, event.Unlink.File.Inode, "wrong inode")
```
Comment (spikat) [inline pkg/security/tests/unlink_test.go:59]: here we could use the new inode assert helper

File: pkg/security/tests/unlink_test.go [hunk 9]
```diff
@@ -81,7 +83,7 @@ func TestUnlink(t *testing.T) {
...
```
Comment (spikat) [inline pkg/security/tests/unlink_test.go:87]: here we could use the new inode assert helper

File: pkg/security/tests/utimes_test.go [hunk 10]
```diff
@@ -56,7 +58,7 @@ func TestUtimes(t *testing.T) {
 			assert.Equal(t, "utimes", event.GetType(), "wrong event type")
 			assert.Equal(t, int64(123), event.Utimes.Atime.Unix())
 			assert.Equal(t, int64(456), event.Utimes.Mtime.Unix())
-			if !test.opts.staticOpts.enableEBPFLess {
+			if !ebpfLessEnabled {
 				assert.Equal(t, getInode(t, testFile), event.Utimes.File.Inode, "wrong inode")
```
Comment (spikat) [inline pkg/security/tests/utimes_test.go:62]: here we could use the new inode assert helper

File: pkg/security/tests/utimes_test.go [hunk 11]
```diff
@@ -97,7 +99,7 @@ func TestUtimes(t *testing.T) {
 			assert.Equal(t, "utimes", event.GetType(), "wrong event type")
 			assert.Equal(t, int64(111), event.Utimes.Atime.Unix())
 			assert.Equal(t, int64(222), event.Utimes.Atime.UnixNano()%int64(time.Second)/int64(time.Microsecond))
-			if !test.opts.staticOpts.enableEBPFLess {
+			if !ebpfLessEnabled {
 				assert.Equal(t, getInode(t, testFile), event.Utimes.File.Inode)
```
Comment (spikat) [inline pkg/security/tests/utimes_test.go:103]: here we could use the new inode assert helper

File: pkg/security/tests/utimes_test.go [hunk 12]
```diff
@@ -141,7 +143,7 @@ func TestUtimes(t *testing.T) {
 			assert.Equal(t, "utimes", event.GetType(), "wrong event type")
 			assert.Equal(t, int64(555), event.Utimes.Mtime.Unix())
 			assert.Equal(t, int64(666), event.Utimes.Mtime.UnixNano()%int64(time.Second)/int64(time.Nanosecond))
-			if !test.opts.staticOpts.enableEBPFLess {
+			if !ebpfLessEnabled {
 				assert.Equal(t, getInode(t, testFile), event.Utimes.File.Inode)
```
Comment (spikat) [inline pkg/security/tests/utimes_test.go:147]: here we could use the new inode assert helper

File: pkg/security/tests/schemas.go [hunk 13]
```diff
@@ -107,6 +107,10 @@ func (tm *testModule) validateExitSchema(t *testing.T, event *model.Event) bool
 
 //nolint:deadcode,unused
 func (tm *testModule) validateOpenSchema(t *testing.T, event *model.Event) bool {
+	if ebpfLessEnabled {
+		return true
+	}
```
Comment (spikat) [inline pkg/security/tests/schemas.go:112]: should make the same update on other schema validation ? at least for those where there is now a "if !ebpflessenabled" to skip them.

File: pkg/security/tests/rename_test.go [hunk 14]
```diff
@@ -64,7 +66,7 @@ func TestRename(t *testing.T) {
 			return nil
 		}, func(event *model.Event, rule *rules.Rule) {
 			assert.Equal(t, "rename", event.GetType(), "wrong event type")
-			if !test.opts.staticOpts.enableEBPFLess {
+			if !ebpfLessEnabled {
 				assert.Equal(t, getInode(t, testNewFile), event.Rename.New.Inode, "wrong inode")
```
Comment (spikat) [inline pkg/security/tests/rename_test.go:70]: could use the new helper here

File: pkg/security/tests/rename_test.go [hunk 15]
```diff
@@ -96,7 +98,7 @@ func TestRename(t *testing.T) {
...
```
Comment (spikat) [inline pkg/security/tests/rename_test.go:102]: could use the new helper here

File: pkg/security/tests/rename_test.go [hunk 16]
```diff
@@ -129,7 +131,7 @@ func TestRename(t *testing.T) {
...
```
Comment (spikat) [inline pkg/security/tests/rename_test.go:135]: could use the new helper here

File: pkg/security/tests/open_test.go [hunk 17]
```diff
@@ -100,14 +93,6 @@ func TestOpen(t *testing.T) {
 	}
 
 	t.Run("openat2", func(t *testing.T) {
-		kv, err := kernel.NewKernelVersion()
-		if err != nil {
-			t.Fatal(err)
-		}
-		if test.opts.staticOpts.enableEBPFLess && kv.Code < kernel.Kernel6_0 {
-			t.Skip("openat2 is not supported")
-		}
```
[the remark this answers is not in the document]
Comment (spikat) [reply pkg/security/tests/open_test.go:109]: Looks on on 5.15

File: pkg/security/tests/module_tester.go (hunk 1 above)
Comment (safchain) [reply pkg/security/tests/module_tester.go:973]: It just simplifies the logic

File: pkg/security/tests/main_test.go [hunk 18]
```diff
@@ -11,14 +11,76 @@ package tests
...
+		available := []string{
+			"~TestProcess",
+			"~TestOpen",
+			"~TestUnlink",
+			"KillAction",
+			"TestRmdir",
+			"TestUnlink",
+			"TestRename",
+			"TestMkdir",
+			"TestUtimes",
```
[the remark this answers is not in the document]
Comment (safchain) [reply pkg/security/tests/main_test.go:50]: That is a good point I'll make them all starting with a ~

File: pkg/security/tests/main_test.go (hunk 4 above)
Comment (safchain) [reply pkg/security/tests/main_test.go:51]: Good catch, thanks

File: pkg/security/tests/module_tester.go (hunk 1 above)
Comment (spikat) [reply pkg/security/tests/module_tester.go:973]: ok right, I looked at it too fast :) sorry

**Comment by spikat** [comment]
When launching the testsuite on the branch, some tests are passing instead of being skipped (missing SkipIfNotAvailable(t)):
```
--- PASS: TestProtobufDecoding (0.00s)
--- PASS: TestFieldsResolver (3.11s)
    --- PASS: TestFieldsResolver/open (0.00s)
    --- PASS: TestFieldsResolver/exec (0.00s)
--- PASS: TestEnv (0.00s)
```

**Review by spikat — APPROVED** [review]
LGTM :) thanks a lot for those improvements

**Review by chouetz — APPROVED** [review]

**Review by YoannGh — APPROVED** [review]

**Pull request merged**